tests/integration/suite/daprd/resiliency/apps/retryfilter.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apps
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(retryfilter))
+}
+
+type retryfilter struct {
+	daprd1   *daprd.Daprd
+	daprd2   *daprd.Daprd
+	counters sync.Map
+}
+
+func (d *retryfilter) Setup(t *testing.T) []framework.Option {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(r.Method))
+	})
+
+	handler.HandleFunc("/retry", func(w http.ResponseWriter, r *http.Request) {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		key := r.URL.Query().Get("key")
+		if key == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		c, _ := d.counters.LoadOrStore(key, &atomic.Int32{})
+		counter := c.(*atomic.Int32)
+		counter.Add(1)
+
+		respStatusCode, err := strconv.Atoi(string(body))
+		if (err != nil) || (respStatusCode < 100) || (respStatusCode >= 600) {
+			// Trying to write a bad status code will cause app to crash.
```

Comment:
To be clear what is crashing here? Nothing should ever be crashing due to this.

Reply:
There is a stacktrace. The app does not crash per se. Updating comment.